app/carbonapi/api.go

```diff
@@ -218,21 +220,21 @@ func StartCarbonapi(api cfg.API, logger *zap.Logger) {
 	}
 }
 
-func registerPromehteusMetrics(logger *zap.Logger) {
+func registerPromehteusMetrics(logger *zap.Logger, app *App) {
```

Comment:
would it make sense for this to be a method on the app? it looks like it uses a fair number of fields, especially `initHandlersInternal`.